lib/puppet/provider/templates/webadministration/logproperties.ps1.erb

```diff
@@ -1,9 +1,44 @@
+$iis_version = '<%= Facter.value(:iis_version) %>'
+
+$W3CValue = 'W3C'
+$MaxSizeValue = 'MaxSize'
+
+# Convert the string representation of an Enum value into an Int32 safely
+Function ConvertStringTo-EnumInt32($EnumType, $Value, $ValueName) {
```

Comment:
This parameter statement makes more sense to me. I was confused by `Value` being specified first, my first instinct is to specify the name of a thing and then it's value.

```
function ConvertTo-EnumInt32 {
  param(
    [string]$EnumType,
    [string]$Name, # formerly $Valuename
    [string]$Value 
    )
}
```

If you want further disambiguation, I could see it as `EnumName`, but I think that's redundant

Then use it:

```
ConvertStringTo-EnumInt32 -EnumType ([Microsoft.Web.Administration.LogFormat]) -Name 'logformat' -Value $logParams.Value